src/components/backend-ai-agent-summary-list.ts

```diff
@@ -0,0 +1,501 @@
+
+        lablup-progress-bar {
+          --progress-bar-width: 70px;
+          border-radius: 3px;
+          height: 10px;
+          --mdc-theme-primary: #3677eb;
+          --mdc-linear-progress-buffer-color: #98be5a;
+          margin-bottom: 0;
+        }
+
+        lablup-progress-bar.cpu {
+          --progress-bar-height: 7px;
+        }
+
+        lablup-progress-bar.cuda {
+          --progress-bar-width: 80px;
+          --progress-bar-height: 15px;
+          margin-bottom: 5px;
+        }
+
+        lablup-progress-bar.mem {
+          --progress-bar-width: 100px;
+          --progress-bar-height: 15px;
+        }
+
+        lablup-progress-bar.utilization {
+          --progress-bar-width: 80px;
+          margin-left: 10px;
+        }
+
+        lablup-shields {
+          margin: 1px;
+        }
+
+        mwc-icon {
+          --mdc-icon-size: 16px;
+        }
+
+        mwc-icon.schedulable {
+          --mdc-icon-size: 24px;
+        }
+
+        vaadin-grid {
+          border: 0;
+          font-size: 14px;
+        }
+    `];
+  }
+
+  firstUpdated() {
+    this.list_status = this.shadowRoot.querySelector('#list-status');
+    this.notification = globalThis.lablupNotification;
+    this.agentDetailDialog = this.shadowRoot.querySelector('#agent-detail');
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  /**
+  * Change state to 'ALIVE' when backend.ai client connected.
+  *
+  * @param {Boolean} active - The component will work if active is true.
+  */
+  async _viewStateChanged(active: boolean) {
+    await this.updateComplete;
+    if (active === false) {
+      return;
+    }
+    // If disconnected
+    if (typeof globalThis.backendaiclient === 'undefined' || globalThis.backendaiclient === null || globalThis.backendaiclient.ready === false) {
+      document.addEventListener('backend-ai-connected', () => {
+        this.enableAgentSchedulable = globalThis.backendaiclient.supports('schedulable');
+        const status = 'ALIVE';
+        this._loadAgent(status);
+      }, true);
+    } else { // already connected
+      this.enableAgentSchedulable = globalThis.backendaiclient.supports('schedulable');
+      const status = 'ALIVE';
+      this._loadAgent(status);
+    }
+  }
+
+  /**
+  * Load an agents list with agent's backend.ai information.
+  *
+  * @param {string} status - The agent's backend.ai client status.
+  */
+  _loadAgent(status = 'running') {
+    if (this.active !== true) {
+      return;
+    }
+    this.list_condition = 'loading';
+    this.list_status.show();
+    const fields = ['id', 'status', 'available_slots', 'occupied_slots', 'architecture'];
+    switch (this.condition) {
+    case 'running':
+      status = 'ALIVE';
+      break;
+    case 'terminated':
+      status = 'TERMINATED';
+      break;
+    case 'archived':
+    default:
+      status = 'ALIVE';
+    }
+
+    if (globalThis.backendaiclient.supports('schedulable')) {
+      fields.push('schedulable');
+    }
+
```

Comment:
Not a big deal, but argument with hard-coded number could be altered with local variables with corresponding value.
For example, `10 * 1000` means timeout.